src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -759,21 +768,150 @@ public boolean isKnownDatasource( UUID studyId, String participantId, String dat
+                        } );
+
+                        // fill Map<entitySetId, Set<entityKeyId>>
+                        participantNeighbors
+                                .get( participantEntityKeyId )
+                                .forEach( ( edgeEntitySetId, edgeNeighbor ) -> {
+                                    edgeNeighbor.forEach( ( neighborEntitySetId, neighborEntityIds ) -> {
+                                        Set<UUID> set = toDeleteEntitySetIdEntityKeyId.get( neighborEntitySetId );
+                                        set.add( neighborEntityIds.getNeighborEntityKeyId() );
+                                    } );
+                                } );
+                    }
+
+            );
+
+            // delete all neighbors
+            toDeleteEntitySetIdEntityKeyId
+                    .forEach(
+                            ( entitySetId, entityKeyId ) -> chronicleDataApi
+                                    .deleteEntities( entitySetId, entityKeyId, deleteType )
+                    );
+
+            // delete participants
+            Integer deleted = userDataApi.deleteEntities( participantsEntitySetId, participantsToRemove, deleteType );
+            logger.info("Deleted {} entities for participant {}.", deleted.toString(), participantId.toString());
+
+            // delete study if no participantId is specified
+            if ( participantId.isEmpty() ) {
+                // delete participant entity set
+                userEntitySetsApi.deleteEntitySet( participantsEntitySetId );
+                logger.info("Deleted participant dataset for study {}.", studyId);
+                UUID studyEntityKeyId = getStudyEntityKeyId( studyId );
+                userDataApi.deleteEntities( studyESID, Set.of( studyEntityKeyId ), deleteType );
+                logger.info("Deleted study {} from global studies dataset.", studyId);
+            }
+
+        } catch ( Exception e ) {
+            String errorMsg = "failed to delete participant data";
+            logger.error( errorMsg, e );
+            throw new RuntimeException( errorMsg );
+        }
+    }
+
     @Override
-    public void deleteParticipantAndAllNeighbors( UUID studyId, String participantId ) {
-        // to do
+    public Void deleteParticipantAndAllNeighbors(
+            UUID studyId,
+            String participantId,
+            DeleteType deleteType,
+            String userToken ) {
+        com.openlattice.data.DeleteType deleteTypeTransformed = com.openlattice.data.DeleteType.valueOf( deleteType.toString() );
+        deleteStudyData( studyId, java.util.Optional.of( participantId ), deleteTypeTransformed, userToken );
+        logger.info( "Successfully removed a participant from {}", studyId );
+
+    return null;
     }
 
     @Override
-    public void deleteStudyAndAllNeighbors( UUID studyId ) {
-        // to do
+    public Void deleteStudyAndAllNeighbors( UUID studyId, DeleteType deleteType, String userToken ) {
```

Comment:
and same for these, since this isn't being called by Retrofit, should use `void`